File: c/fredex.c
#include <stdio.h>

int main(int argc, char *argv[])
{
	int minWeight, cost, weight, bestCost = 0x7FFFFFFF;
	FILE *in = fopen(argv[1], "r");
	
	fscanf(in, "Unload: %d\n", &minWeight);
	
	while(fscanf(in, "%d %d\n", &cost, &weight) > 0)
		bestCost = (((minWeight / weight) + 1) * cost < bestCost) ? ((minWeight / weight) + 1) * cost : bestCost;
	
	printf("%d\n", bestCost);
	return 0;
}

File: lua/fredex.lua
local minWeight, cost, weight, totalCost;
local bestCost = 200000000;
file = io.open(arg[1], 'r');

file:read(8);
minWeight = file:read('*n');

while true do
	cost = file:read('*n');
	weight = file:read('*n');
	if cost == nil or weight == nil then break; end
	totalCost = math.floor(minWeight / weight) * cost;
	if(minWeight % weight ~= 0) then totalCost = totalCost + cost; end
	if(totalCost < bestCost) then bestCost = totalCost; end
end

print(bestCost);
